internal/usecase/daemon/daemon.go

```
@@ -1105,90 +1134,73 @@ func (d *Daemon) handleList(tr ports.Transport) {
 	stopTransport := watchHandshakeTransport(handshakeCtx, tr)
 	defer finishHandshake()
 	defer stopTransport()
-	d.handleHelloWithContext(handshakeCtx, timedOut, stopTransport, finishHandshake, tr, f)
+	d.handleHelloWithContext(handshakeCtx, timedOut, stopTransport, finishHandshake, tr, hello)
 }
 
 // handleHelloWithContext runs the attach handshake: version check, intent
 // routing, Welcome, guaranteed first paint, then the per-connection input loop.
-func (d *Daemon) handleHelloWithContext(handshakeCtx context.Context, timedOut <-chan struct{}, stopHandshakeTransport, finishHandshake func(), tr ports.Transport, f wire.Frame) {
+func (d *Daemon) handleHelloWithContext(handshakeCtx context.Context, timedOut <-chan struct{}, stopHandshakeTransport, finishHandshake func(), tr ports.ServerConnection, h protocol.Hello) {
 	if err := handshakeContextError(context.Background(), timedOut, handshakeCtx.Err()); err != nil {
```

golangci-lint (2.12.2)

[error] Non-inherited new context, use function like `context.WithXXX` instead

(contextcheck)